src/back/hlsl/writer.rs

```diff
@@ -1227,6 +1244,79 @@ impl<'a, W: Write> Writer<'a, W> {
                     write!(self.out, ")")?;
                 }
             }
+            Expression::ImageLoad {
+                image,
+                coordinate,
+                array_index: _, // TODO:
+                index: _,       // TODO:
+            } => {
+                // https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-to-load
+                let image_ty = func_ctx.info[image].ty.inner_with(&module.types);
+                if let crate::TypeInner::Image {
```

Comment:
similarly to other places, we should just bail out (with `unreachable` or `continue`) if the match fails instead of indenting this much code here